code/modules/integrated_electronics/core/printer.dm

```diff
@@ -16,11 +21,28 @@
 /obj/item/device/integrated_circuit_printer/upgraded
 	upgraded = TRUE
 	can_clone = TRUE
+	fast_clone = TRUE
 
 /obj/item/device/integrated_circuit_printer/Initialize()
 	. = ..()
 	AddComponent(/datum/component/material_container, list(MAT_METAL), MINERAL_MATERIAL_AMOUNT * 25, TRUE, list(/obj/item/stack, /obj/item/integrated_circuit, /obj/item/device/electronic_assembly))
 
+/obj/item/device/integrated_circuit_printer/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/obj/item/device/integrated_circuit_printer/process()
+	if(!cloning)
+		STOP_PROCESSING(SSprocessing, src)
+	clone_countdown--
+	if(!clone_countdown || fast_clone)
```

Comment:
this can be achieved in a much more performant way using stoppable timers

Reply:
Would that be by using something like
`clone_countdown = addtimer(blah etc. blah)`
and then deleting that timer if canceled?